app/src/main/java/org/systers/mentorship/view/activities/LoginActivity.kt

```diff
@@ -2,131 +2,234 @@ package org.systers.mentorship.view.activities
 import org.systers.mentorship.R
-import org.systers.mentorship.remote.requests.Login
 import org.systers.mentorship.utils.Constants
-import org.systers.mentorship.utils.CountingIdlingResourceSingleton
+import org.systers.mentorship.view.CircularIndeterminateProgressBar
+import org.systers.mentorship.view.DefaultSnackBar
+import org.systers.mentorship.view.EmailTextField
+import org.systers.mentorship.view.PasswordTextField
 import org.systers.mentorship.viewmodels.LoginViewModel
 
 /**
  * This activity will let the user to login using username/email and password.
  */
 class LoginActivity : BaseActivity() {
     private val loginViewModel: LoginViewModel by viewModels()
-    private lateinit var username: String
-    private lateinit var password: String
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_login)
 
-        etUsername.addTextChangedListener(textWatcher)
-        etPassword.addTextChangedListener(textWatcher)
+        setContent {
+            LoginScreen(viewModel = loginViewModel,
+                onSignUpButtonClick = {
+                    intent = Intent(this, SignUpActivity::class.java)
+                    startActivity(intent)
+                })
+        }
 
-        loginViewModel.successful.observe(this, {
-            successful ->
+        loginViewModel.successful.observe(this) { successful ->
             hideProgressDialog()
             if (successful != null) {
                 if (successful) {
                     Toast.makeText(this, R.string.logging_successful, Toast.LENGTH_LONG)
-                            .show()
+                        .show()
                     intent = Intent(this, MainActivity::class.java)
                     startActivity(intent)
                     finish()
-                } else {
-                    Snackbar.make(getRootView(), loginViewModel.message, Snackbar.LENGTH_LONG)
-                            .show()
                 }
-                CountingIdlingResourceSingleton.decrement()
             }
-        })
-
-        btnLogin.setOnClickListener {
-            login()
-        }
-
-        btnSignUp.setOnClickListener {
-            intent = Intent(this, SignUpActivity::class.java)
-            startActivity(intent)
-        }
-
-        tiPassword.editText?.setOnEditorActionListener { _, actionId, _ ->
-            if (actionId == EditorInfo.IME_ACTION_DONE) {
-                login()
-            }
-            false
         }
 
         try {
             val tokenExpiredVal = intent.extras!!.getInt(Constants.TOKEN_EXPIRED_EXTRA)
             if (tokenExpiredVal == 0)
-                Snackbar.make(getRootView(), "Session token expired, please login again", Snackbar.LENGTH_LONG).show()
-        } catch (exception: Exception) {}
-
-        checkFieldsForEmptyValues()
+                Toast.makeText(this, "Session Token expired, please login again", Toast.LENGTH_LONG)
```

Comment:
you created snackbar but using toast? I think there are new contextual toast, so you know which activity it is related to - not sure if this is not part of Android 12 though.

Reply:
I used a toast because that was just the way it was implemented before. I agree and think we can move this to a snackbar.